src/Microsoft.Diagnostics.Runtime/src/Linux/Structs/ElfFileTableEntryPointers64.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Microsoft.Diagnostics.Runtime.Linux
+{
+    [StructLayout(LayoutKind.Sequential)]
+    public struct ElfFileTableEntryPointers64
+    {
+        public long Start;
```

Comment:
I agree that all the fields should be unsigned.  With a couple of exceptions, all the C++ ELF data structure fields are unsigned.  The only reason I left some of them signed is that various "position" arguments to "read" functions in the infrastructure are "long" and it would require a lot of changes.

I'll make all these changes because you never know when signed problems will hit. Like if you have > 2GB core dumps.

Reply:
I've made all the ELF structs unsigned except in a couple of cases in the prstatus that the linux header defines as signed. But the wrappers (IELFHeader, ElfProgramHeader) still expose the offsets and addresses as "long". 

The changes to make these addresses, offsets ("position") and sizes unsigned (ulong) all the way through the ELF dump reader is way too many changes. It may be the right thing, but it really gets complicated. Both the address and size need to be unsigned because they are combined in a lot of calculations.  The IAddressSpace.Read implementation in StreamAddressSpace calls Stream.Seek which takes a long and "Length" on this interface returns Stream.Length which is long. So I left all the lower layers alone.

Another complication is that I had to use "unchecked" (used checked in all the other conversions) on the ElfProgramHeader.VirtualAddress/VirtualSize conversion because there are some LOAD records with VirtualAddress = 0xffffffffff600000.  Converting this to long may cause problems in the math/compares in CoreDumpReader.VirtualQuery. 

It may be simpler to have a short discussion about this if you want.

Reply:
My two concerns are:

1.  It should work.  (Sounds like it does.)
2.  Don't expose any signed (`long` or `IntPtr`) fields or parameters that are addresses.  This is inconsistent with the api design of clrmd.  (Offhand I don't see any public class or struct doing this, but it's a big change.)

As long as those two things are satisfied it's ok if we need to do some ugly math in the guts of the code.  This isn't the first time we've had to deal with it in ClrMD.